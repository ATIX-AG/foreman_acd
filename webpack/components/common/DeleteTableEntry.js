import React from 'react';
import PropTypes from 'prop-types';
import {
  Icon,
  Button,
} from 'patternfly-react';
import { translate as __ } from 'foremanReact/common/I18n';

const DeleteTableEntry = ({
  hidden,
  disabled,
  onDeleteTableEntry,
  additionalData,
}) =>{
  if (hidden) {
    return null;
  }

  return (
    <span>
      <Button
        bsStyle="default"
        disabled={disabled}
        onClick={() => window.confirm(__("Are you sure you wish to delete this item?")) && onDeleteTableEntry(additionalData) }
      >
        <Icon type="pf" name="delete" title={__("Delete entry")} />
      </Button>
    </span>
  );
};

DeleteTableEntry.propTypes = {
  hidden: PropTypes.bool.isRequired,
  disabled: PropTypes.bool.isRequired,
  onDeleteTableEntry: PropTypes.func.isRequired,
  additionalData: PropTypes.object.isRequired,
};

export default DeleteTableEntry;
