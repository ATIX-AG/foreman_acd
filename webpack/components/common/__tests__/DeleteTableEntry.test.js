import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import DeleteTableEntry from '../DeleteTableEntry';

const noop = () => {};

const fixtures = {
  'should render delete parameter': {
    hidden: false,
    disabled: false,
    onDeleteTableEntry: noop,
    additionalData: {},
  },
  'should render hidden button': {
    hidden: true,
    disabled: false,
    onDeleteTableEntry: noop,
    additionalData: {},
  },
  'should render disabled button': {
    hidden: false,
    disabled: true,
    onDeleteTableEntry: noop,
    additionalData: {},
  },
};

describe('DeleteTableEntry', () =>
  testComponentSnapshotsWithFixtures(DeleteTableEntry, fixtures));
