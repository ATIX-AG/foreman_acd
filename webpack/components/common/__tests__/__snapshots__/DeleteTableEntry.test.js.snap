// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteTableEntry should render delete parameter 1`] = `
<span>
   
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={false}
    onClick={[Function]}
  >
    <Icon
      name="delete"
      type="pf"
    />
  </Button>
</span>
`;

exports[`DeleteTableEntry should render disabled button 1`] = `
<span>
   
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={true}
    onClick={[Function]}
  >
    <Icon
      name="delete"
      type="pf"
    />
  </Button>
</span>
`;

exports[`DeleteTableEntry should render hidden button 1`] = `""`;
